Assets/MixedRealityToolkit/Providers/Hands/HandRay/TorsoDirectedHandRay.cs

```diff
@@ -0,0 +1,155 @@
+﻿using Microsoft.MixedReality.Toolkit.Utilities;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    /// <summary>
+    /// Alternate hand ray implementation by prvncher (Eric Provencher)
+    ///
+    /// What does this do?
+    /// - Rather than computing hand ray direction by estimating shoulder position by offsetting a point using the head rotation,
+    /// we estimate a torso direction, using head and hand positions, and use that to estimate the shoulder offset.
+    /// 
+    /// Why use this class instead of the original hand ray?
+    /// - More resilient to head rotating while working with menu.
+    /// - Overall more stable, with a more comfortable angle.
+    ///
+    /// How to use this class?
+    /// - Right now, there is no mechanism in place to introduce an alternate hand ray implementation.
+    /// - For now, to get this working, simply replace the class HandRay, by TorsoDirectedHandRay in BaseHand.cs.
+    /// </summary>
+    public class TorsoDirectedHandRay : IHandRay
+    {
+        /// <inheritdoc />
+        public Ray Ray
+        {
+            get
+            {
+                ray.origin = stabilizedRay.StabilizedPosition;
+                ray.direction = stabilizedRay.StabilizedDirection;
+                return ray;
+            }
+        }
+
+        /// <inheritdoc />
+        public bool ShouldShowRay
+        {
+            get
+            {
+                Vector3 projectedPalm = Vector3.ProjectOnPlane(palmNormal.normalized, headUp);
+
+                bool forwardAcceptable = Vector3.Dot(projectedPalm, headForward) > CursorBeamBackwardTolerance;
+
+                return forwardAcceptable;
+            }
+        }
+
+        private Ray ray = new Ray();
+
+        // Beam alignment check.
+        private const float CursorBeamBackwardTolerance = 0.3f;
+
+        // Angle to curve the hand ray inwards with (degrees).
```

Comment:
consider using /// for those comments - that way they would show up in the api docs